Started Event API folder and latest yaml

nissimsan opened the pull request:
Hey guys - let me know if this is along the lines of how you think this should be structured. Added a folder for our work and the latest yaml. There's more documentation to come, just wanted to get it started now that we are all focusing on getting git up and running... 
Nis

nissimsan committed:
Started Event API folder and latest yaml

petesavitsky reviewed (commented):
Hey Nis,

This looks cool! Just a couple thoughts on repo structure:

Maybe we can add an `events-api` folder in the top level directory? Then within that, an `events.md` and then a maybe a swagger subfolder where you can put the swagger yml? I'm assuming that's what this is.

 So something like: 
- `event-api/event-api.md`
- `event-api/swagger/event-api.yaml`

The event-api.md could be similar to the other pages on standards and charter, in that it would explain what this is and what it's for.

nissimsan commented:
For sure, we'll definitely add an .md! 
I'll move the yaml into a swagger subfolder.

nissimsan committed:
Started an MD and moved yaml under /swagger

nissimsan committed:
We've found that you have to include both Locode and SMDG code. This is
because sometimes the terminal is part of a port different from its
location. For example, some New Jersey terminals are part of the port of
New York. Thus, this isn't a perfect hierarchy, but rather the (correct)
representation of that terminal would be:
UNLocode: USNJN
SMDG: USNYC-AMPT
... So we've decided to concatenate with a dash in between. Hope you're
okay on that? I doubt it breaks any of your stuff, nyshex...?

nissimsan commented:
Added a small change on SMDG too, @petesavitsky. It's something we've discussed in a few meetings. Let me know how you find it...

nissimsan commented:
(I really need to make some branches - this is getting a bit ugly)  :)

nissimsan committed:
Adding more text...

nissimsan committed:
Adding more text...

nissimsan committed:
image

nissimsan committed:
goofing around

nissimsan committed:
More documentation, incl images

nissimsan committed:
Fixed image ref

nissimsan committed:
More images

nissimsan committed:
More text

nissimsan committed:
Missing /images

nissimsan commented:
Alright, done with a baseline of the documentation now...

rgenoulaz commented:
Looks like it's becoming real.
This is awesome, thank you guys! 
Let's merge this and move forward.

Cheers

Romain

rgenoulaz reviewed (approved):
Lets's go with this first version !
Thanks

petesavitsky commented on event-api/event-api.md:
```diff
@@ -0,0 +1,126 @@
+
+### UN/CEFACT Foundation
+![UNCEFACT Buy-Ship-Pay](images/cefact-bsp.png)
+The Event API allows for publishing Consignments and Transport Equipment as events through a POSTing alongside the events, meant to serves as the context of those subsequent events. 
```
`...meant to serve[s] as the context...`

petesavitsky commented on event-api/event-api.md:
```diff
@@ -0,0 +1,126 @@
+These form a hierarchy, with Transport Equipment sitting underneath a Consignment. Consignment also hold parties. Planned events relate to Consignments; Estimated and Actual events relate to Transport Equipment. 
+From the perspective of common carrier business processes, the Consignment roughly represents the booking confirmation, whereas the Transport Equipment represents the equipment matching (when a physically identified container has been assigned to the consignment). 
+
+Shipment plays an important role, but has not yet been included in this first version of the API entirely due to scope confinement. The Shipment class represents the traded of goods inside the containers, and are of important relevance to authorities, financial institutions, sellers and sellers. It is worth noting that LCL tracking should be done through the Shipment class, so the current version ...
```
`...represents the [traded of goods?] inside the containers...`

petesavitsky reviewed (approved):
@nissimsan looks good, just a few potential typos in the .md. It would also be cool to stick a link to the swagger API in the .md, I'm assuming you're using an "official" one?
